pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/RealtimeSegmentAssignment.java

```diff
@@ -158,7 +153,8 @@ private List<String> assignConsumingSegment(String segmentName, InstancePartitio
       int numInstances = instances.size();
       List<String> instancesAssigned = new ArrayList<>(_replication);
       for (int replicaId = 0; replicaId < _replication; replicaId++) {
-        instancesAssigned.add(instances.get((partitionGroupId * _replication + replicaId) % numInstances));
+        int instanceIndex = Math.abs(partitionGroupId * _replication + replicaId) % numInstances;
```

Comment:
Does `getPartitionGroupId()` aim to return a non-negative value? If so, we don't need Math.abs() here.

Reply:
If partitionGroupId is a large number, `partitionGroupId * _replication + replicaId` will roll over to negative numbers. That's why I put the Math.abs() around it. Having said that, if we bound the partitionGroupId to a smaller number - like % 1000 that Jackie is suggesting - we can remove the Math.abs() here.